distributed/shuffle/tests/test_shuffle.py

```diff
@@ -952,9 +916,9 @@ def block(df, in_event, block_event):
 
         await in_event.wait()
         await n.process.process.kill()
-        block_event.set()
-        with pytest.raises(RuntimeError):
-            await fut
+        await block_event.set()
+
+        await fut
```

Comment:
If this is now able to recover we should add a couple of additional asserts

1. Assert that there is indeed a shuffle task on the dead worker
2. Assert that the output result is what is expected

Reply:
I'm surprised that this change allows us to rerun. Do you understand why this works? You also reduced the data / number of partitions which makes me nervous in thinking that the now-dead worker just didn't run any shuffle tasks

Reply:
thanks for digging in